### PR TITLE
[AIRFLOW-3780] Fix some pages are incorrectly directly after set base_url

### PR DESCRIPTION
**Pull request by yiheng**
Make sure you have checked all steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3780
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

We found that after set the base_url, some of the aiflow pages cannot be redirected  to correct path. The pages include:
* Task instances in the dag detail page
* The schedule interval tag in the dag detail page
* Trigger button in the dag list page

In the first two pages, the root cause is the path is hard coded. In the PR, I change them to {{ url_for( ... ) }} to use the path generated by backend.

In the third one, the root cause is that the 'origin' param is not passed to the backend correctly. I added a javascript function to add the origin param to the url.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

I don't know how to do test on UI link change

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`
wait for travis

### CODE REVIEW

File: airflow/www/templates/airflow/dags.html [hunk 1]
```diff
@@ -477,5 +477,9 @@ <h2>DAGs</h2>
           container: "body",
         });
       });
+      function urlWithOrigin(url) {
+        url = url + '&origin=' + encodeURIComponent(window.location);
+        window.location = url;
```
Comment (ashb) [inline airflow/www/templates/airflow/dags.html]: What's going on here - why do we need this function and to change the link from a normal one to one requiring Javascript?

Comment (ashb) [reply airflow/www/templates/airflow/dags.html]: Can we use [`request.path`](http://flask.pocoo.org/docs/1.0/api/#flask.Request.path) instead?

Comment (yiheng) [reply airflow/www/templates/airflow/dags.html]: Looks like `origin={{ request.base_url }}` fits here. Thanks for your suggestion. I'm going to change it.

**Comment by yiheng** [comment]
Have changed the code based on review comments. Please help take a look, thanks! @ashb

**Pull request merged**